frontend/javascripts/oxalis/controller/combinations/tool_controls.js

```diff
@@ -0,0 +1,271 @@
+        }
+      },
+      over: () => {
+        MoveHandlers.handleOverViewport(planeId);
+      },
+      pinch: delta => MoveHandlers.zoom(delta, true),
+      mouseMove: (delta: Point2, position: Point2, id, event) => {
+        // Always set the correct mouse position. Otherwise, using alt + mouse move and
+        // alt + scroll won't result in the correct zoomToMouse behavior.
+        MoveHandlers.setMousePosition(position);
+        if (event.altKey && !event.shiftKey) {
+          MoveHandlers.handleMovePlane(delta);
+        }
+      },
+      leftDownMove: (delta: Point2, _pos: Point2, _id: ?string, _event: MouseEvent) => {
+        MoveHandlers.handleMovePlane(delta);
+      },
+      middleDownMove: MoveHandlers.handleMovePlane,
+      rightClick: MoveTool.createRightClickHandler(planeView, showNodeContextMenuAt),
+    };
+  }
+
+  static createRightClickHandler(
+    planeView: PlaneView,
+    showNodeContextMenuAt: ShowContextMenuFunction,
+  ) {
+    return (pos: Point2, plane: OrthoView, event: MouseEvent, isTouch: boolean) =>
+      SkeletonHandlers.openContextMenu(
+        planeView,
+        pos,
+        plane,
+        isTouch,
+        event,
+        showNodeContextMenuAt,
+      );
+  }
+}
+
+export class SkeletonTool {
+  static getMouseControls(planeView: PlaneView, showNodeContextMenuAt: ShowContextMenuFunction) {
+    return {
+      leftDownMove: (delta: Point2, pos: Point2, _id: ?string, event: MouseEvent) => {
+        const { tracing } = Store.getState();
+        if (tracing.skeleton != null && event.ctrlKey) {
+          SkeletonHandlers.moveNode(delta.x, delta.y);
+        } else {
+          MoveHandlers.handleMovePlane(delta);
+        }
+      },
+      leftClick: (pos: Point2, plane: OrthoView, event: MouseEvent, isTouch: boolean) =>
+        this.onClick(planeView, pos, event.shiftKey, event.altKey, event.ctrlKey, plane, isTouch),
+      rightClick: (position: Point2, plane: OrthoView, event: MouseEvent, isTouch: boolean) => {
+        const { activeViewport } = Store.getState().viewModeData.plane;
+        if (activeViewport === OrthoViews.TDView) {
+          return;
+        }
+
+        if (event.shiftKey) {
+          SkeletonHandlers.handleOpenContextMenu(
+            planeView,
+            position,
+            plane,
+            isTouch,
+            event,
+            showNodeContextMenuAt,
+          );
+        } else {
+          SkeletonHandlers.handleCreateNode(planeView, position, event.ctrlKey);
+        }
+      },
+      middleClick: (pos: Point2, plane: OrthoView, event: MouseEvent) => {
+        if (event.shiftKey) {
+          handleAgglomerateSkeletonAtClick(pos);
+        }
+      },
+    };
+  }
+
+  static onClick(
+    planeView: PlaneView,
+    position: Point2,
+    shiftPressed: boolean,
+    altPressed: boolean,
+    ctrlPressed: boolean,
+    plane: OrthoView,
+    isTouch: boolean,
+  ): void {
+    if (!shiftPressed && !isTouch && !ctrlPressed) {
```

Comment:
I think it would be easier to understand if the `else` branch further down would be an `else if` and the branch conditions would show the full conditions that's needed. For example, for the handleMergeTrees alt **and shift** (or ctrl would actually work as well?!) need to be pressed which is very hard to see right now.

Reply:
Very good point! Historically grown, I guess. I rewrote the logic per your suggestion and also noticed that one can select a node with a mere tap (i.e., touch). Wasn't that obvious, either, when reading the code.
I pushed the changes to #5422. See 605e044a1. Probably also fixed that ctrl and alt could also merge trees.